### DISCUSSION TITLE
`useFetcher` breaks the client side routing

### DISCUSSION THREAD

**Discussion by abereghici** [remix-run/remix#4804] [Q&A]
### What version of Remix are you using?

1.1.1

### What version of Node are you using? Minimum supported version is 14.

v14.15.4

### Steps to Reproduce

I created an abstraction for remix themes (https://github.com/abereghici/remix-themes) and in the theme provider I used useFetcher which caused the page to be reloaded each time I clicked to links. I replaced useFetcher with a simple fetch and now is working as expected. What might be wrong here ? https://github.com/abereghici/remix-themes/blob/main/src/theme-provider.tsx#L60

```

### Expected Behavior

`useFetcher` won't break the client side routing.

### Actual Behavior

`useFetcher` breaks the client side routing. Every link reloads the page.

**Comment by Ehesp** [comment]
Do you have the Scripts component in your root?

**Comment by abereghici** [comment]
Yes, I have. I upgraded from 1.0.6 version and I encountered this problem. Also the `intent=prefetch` for the links reloads the page entirely.

**Comment by abereghici** [comment]
I created an abstraction for remix themes (https://github.com/abereghici/remix-themes) and in the theme provider I used `useFetcher` which caused the page to be reloaded each time I clicked to links. I replaced `useFetcher` with a simple `fetch` and now is working as expected. What might be wrong here ? https://github.com/abereghici/remix-themes/blob/main/src/theme-provider.tsx#L60

**Comment by geocine** [comment]
I suggest you should update the original issue you raised and mention `useFetcher` specifically
